lib/couchdb.js

```diff
@@ -1276,51 +1159,42 @@ CouchDB.prototype._find = function(dbName, query, cb) {
 };
 
 /**
- * Create an index in couchdb,
- * you can specify the ddocName and indexName,
- * regarding how to create model/property index, see discussion in
- * https://github.com/strongloop/loopback-connector-cloudant/pull/153#issuecomment-307177910
- * @param {String} ddocName without prefix `_design/`
- * @param {String} indexName index name
- * @param {Array} fields we only allow single field atm,
- * so it should be a string actually
+ * A model index's naming convertion: '_design/LBModel__Foo__LBIndex__foo_index',
+ * this function returns the model prefix, default as 'LBModel'.
+ * 
+ * TBD: Do we allow customized prefix? same question for 
+ * `getIndexPropertyPrefix`, allow the flexibility would cause 
+ * potential race condition, e.g. two models have same index's 
+ * ddoc name
  */
-CouchDB.prototype.createIndex = function(ddocName, indexName, fields, cb) {
-  debug('createIndex: ddocName %s, indexName %s, fields %s', ddocName,
-    indexName, fields);
-
-  var self = this;
-  var indexBody = {
-    index: {
-      fields: fields,
-    },
-    ddoc: ddocName,
-    name: indexName,
-    type: 'json',
-  };
-
-  var requestObject = {
-    db: self.settings.database,
-    path: '_index',
-    method: 'post',
-    body: indexBody,
-  };
-
-  self.couchdb.request(requestObject, cb);
-};
+CouchDB.prototype.getIndexModelPrefix = function(mo) {
+  const DEFAULT_MODEL_PREFIX = 'LBModel';
+  return DEFAULT_MODEL_PREFIX;
+}
 
 /**
- * Get all indexes in a database.
+ * A model instance is stored with the property specify which model it belongs
+ * to, e.g. {name: 'foo', pwd: 'bar', loopback__model__name: 'User'}
+ * This function returns the default model view: 'loopback__model__name'
  */
-CouchDB.prototype.getIndexes = function(cb) {
-  var self = this;
-  var requestObject = {
-    db: self.settings.database,
-    path: '_index',
-    method: 'get',
-  };
+CouchDB.prototype.defaultModelView = function(){
+  const DEFAULT_MODEL_PREFIX = 'loopback__model__name';
+  return DEFAULT_MODEL_PREFIX;
+}
 
-  self.couchdb.request(requestObject, cb);
-};
+/** 
+ * A model index's naming convertion: '_design/LBModel__Foo__LBIndex__foo_index',
+ * this function returns the property prefix, default as 'LBIndex'.
+ * Default as 'LBIndex'
+ */
+CouchDB.prototype.getIndexPropertyPrefix = function(model) {
+  const DEFAULT_PROPERTY_PREFIX = 'LBIndex';
+  return DEFAULT_PROPERTY_PREFIX;
+}
+
+// mixins
+// require('./discovery')(CouchDB);
```

Comment:
I copy paste the old code, just moved them to different lines, while git diff doesn't show that :(